### PR TITLE
Handle error states

### PR DESCRIPTION
**Pull request by mijiga**
#44 
- Handles errors on home screen e.g. Isar initialisation, `deleteAccount()`, `getUserAccounts()` e.t.c
- Handles `addAccount` on add account screen.
- Refactors out account list and search results from home screen into their own widgets and files

### CODE REVIEW

**Comment by mijiga** [comment]
@zikani03 @Metro13 kindly review

File: app/lib/data/cubit/accounts_cubit.dart [hunk 1]
```diff
@@ -6,13 +7,15 @@ part 'accounts_state.dart';
 
 class AccountsCubit extends Cubit<AccountsState> {
   IsarRepo repository = IsarRepo.instance;
+  List<Account> accounts = [];
```
Comment (zikani03) [inline app/lib/data/cubit/accounts_cubit.dart:10]: Is there a good reason why we should store the accounts here?

File: app/lib/presentation/widgets/account_list_widget.dart [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import 'package:bankimoon/presentation/widgets/card.dart';
+import 'package:flutter/material.dart';
+
+class AccountListWidget extends StatelessWidget {
+  final List accounts;
```
Comment (zikani03) [inline app/lib/presentation/widgets/account_list_widget.dart:5]: make this `List<Account>`

**Review by zikani03 — CHANGES_REQUESTED** [review]

File: app/lib/data/cubit/accounts_cubit.dart (hunk 1 above)
Comment (mijiga) [reply app/lib/data/cubit/accounts_cubit.dart:10]: Yes. Firstly, my goal here was to ensure error states don't replace the entire UI just to communicate an error. Inorder to achieve that the account list has to be persisted across states and putting it there does just that. 

An even better placement would be in the states themselves as a payload but that requires alot more changes than the scope of this PR allows.

Comment (zikani03) [reply app/lib/data/cubit/accounts_cubit.dart:10]: Alright, makes sense.

**Review by zikani03 — APPROVED** [review]

**Pull request merged**